velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -167,19 +177,55 @@ class S3FileSystem::Impl {
     }
   }
 
-  // Configure default AWS credentials provider chain.
+  // Configure an AWSCredentialsProvider with access key and secret key.
+  std::shared_ptr<Aws::Auth::AWSCredentialsProvider>
+  getAccessKeySecretKeyCredentialsProvider(
+      const std::string& accessKey,
+      const std::string& secretKey) const {
+    return std::make_shared<Aws::Auth::SimpleAWSCredentialsProvider>(
+        awsString(accessKey), awsString(secretKey));
+  }
+
+  // Configure default AWSCredentialsProvider.
   std::shared_ptr<Aws::Auth::AWSCredentialsProvider>
   getDefaultCredentialProvider() const {
     return std::make_shared<Aws::Auth::DefaultAWSCredentialsProviderChain>();
   }
 
-  // Configure with access and secret keys.
+  // Configure an AWSCredentialsProvider with S3 IAM Role.
   std::shared_ptr<Aws::Auth::AWSCredentialsProvider>
-  getAccessSecretCredentialProvider(
-      const std::string& accessKey,
-      const std::string& secretKey) const {
-    return std::make_shared<Aws::Auth::SimpleAWSCredentialsProvider>(
-        awsString(accessKey), awsString(secretKey), awsString(""));
+  getIAMRoleCredentialsProvider(
+      const std::string& s3IAMRole,
+      const std::string& sessionName) const {
+    return std::make_shared<Aws::Auth::STSAssumeRoleCredentialsProvider>(
+        awsString(s3IAMRole), awsString(sessionName));
+  }
+
+  // Return an AWSCredentialsProvider based on the config.
+  std::shared_ptr<Aws::Auth::AWSCredentialsProvider> getAWSCredentialsProvider()
```

Comment:
I mostly emulated the Java implementation
https://github.com/prestodb/presto/blob/b7f31ded9cdd68ff6486941fc7e0a4de260c3113/presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java#L816